app/src/main/java/org/dhis2/usescases/settings/SyncManagerFragment.java

```diff
@@ -247,6 +215,60 @@ public void showSmsSettings(boolean enabled, String number, boolean waitForRespo
                 .setText(responseSender);
         ((TextView) binding.settingsSms.findViewById(R.id.settings_sms_result_timeout))
                 .setText(Integer.toString(timeout));
+
+        setSMSListeners();
+    }
+
+    private void setSMSListeners(){
+
+        listenerDisposable.add(RxTextView.textChanges(binding.settingsSms.findViewById(R.id.settings_sms_receiver))
+                .debounce(1000, TimeUnit.MILLISECONDS, Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        data -> {
+                            if (!isEmpty(data)) {
+                                presenter.smsNumberSet(data.toString());
+                            }
+                        },
+                        Timber::d
+                ));
+
+        listenerDisposable.add(RxCompoundButton.checkedChanges(binding.settingsSms.findViewById(R.id.settings_sms_switch))
+                .debounce(1000, TimeUnit.MILLISECONDS, Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        isChecked ->
+                                presenter.smsSwitch(
+                                        NetworkUtils.isOnline(context) &&
+                                                isChecked &&
+                                                checkSMSPermissions()
```

Comment:
shouldn't this be !NetworkUtils.isOnline(context) ?

Reply:
No, you need to be online to enable sms as it will download some extra data